tools/copy_bom/src/util.rs

```diff
@@ -495,3 +531,70 @@ fn deal_with_output(output: Output, error_number: i32) {
         std::process::exit(error_number);
     }
 }
+
+/// Add missing library to a global hashset
+fn add_missing_library(missing_library: String) {
+    MISSING_LIBRARIES
+        .lock()
+        .expect("Acquire lock should not fail")
+        .insert(missing_library);
+}
+
+/// Check whether missing library exists in image directory
+fn check_missing_library(missing_library: &str, image_dir: &str) -> bool {
+    let mut command = Command::new("find");
+    command.arg(image_dir).arg("-name").arg(missing_library);
+    let output = command.output().expect("find missing library failed");
+    let stdout = String::from_utf8_lossy(&output.stdout).to_string();
+    if stdout.len() == 0 {
+        return false;
+    } else {
+        let mut lines = stdout.lines();
+        debug!(
+            "find missing library {} in {}",
+            missing_library,
+            lines.nth(0).unwrap()
+        );
+        return true;
+    }
+}
+
+/// check whether missing libraries exist in image dir
+pub fn lazy_check_missing_libraries(image_dir: &str) {
+    let mut check_failed_missing_libraries = HashSet::new();
+    MISSING_LIBRARIES
+        .lock()
+        .unwrap()
+        .iter()
+        .for_each(|missing_library| {
+            if !check_missing_library(missing_library, image_dir) {
+                check_failed_missing_libraries.insert(missing_library.to_owned());
+            }
+        });
+    if check_failed_missing_libraries.len() > 0 {
+        println!("copy_bom failed due to following libraries are missing:");
+        for library in check_failed_missing_libraries {
+            println!("{}", library);
+        }
+        std::process::exit(MISSING_LIBRARY_ERROR);
+    }
+}
+
+/// Print warn message if image directory is not empty.
+/// The image dir should be empty before running copy_bom so copy bom can track all files to copy in bom file.
+/// Users can ignore the warning now if they are sure the image directory contains correct contents.
+pub fn warn_on_nonempty_image_dir(image_dir: &str) {
+    let image_path_buf = PathBuf::from(image_dir);
+    let image_path = image_path_buf.as_path();
+    if image_path.is_dir() {
+        if let Ok(read_dir) = image_path.read_dir() {
+            let dir_entries = read_dir.collect::<Vec<_>>();
+            if dir_entries.len() > 0 {
+                warn!(
```

Comment:
So only if the user enables the log messages would appear.

Reply:
Fixed.